src/apps/advanced-search-v2/hooks/use-paginated-results.ts

```diff
@@ -0,0 +1,138 @@
+import { useState, useEffect, useMemo } from "react";
+import usePager from "../../../components/result-pager/use-pager";
+import { Work } from "../../../core/utils/types/entities";
+import {
+  useComplexSearchWithPaginationQuery,
+  HoldingsStatusEnum
+} from "../../../core/dbc-gateway/generated/graphql";
+
+export interface UsePaginatedResultsReturn {
+  resultItems: Work[];
+  hitcount: number;
+  isLoading: boolean;
+  isFetching: boolean;
+  isRefetching: boolean;
+  canShowZeroResults: boolean;
+  page: number;
+  PagerComponent: React.FC<{ isLoading?: boolean }>;
+  isLoadingOrRefetching: boolean;
+  shouldShowSearchResults: boolean;
+  shouldShowResultHeadline: boolean;
+  shouldShowZeroResults: boolean;
+}
+
+interface UsePaginatedResultsProps {
+  cql: string;
+  hasQuery: boolean;
+  onShelf: boolean;
+  pageSize: number;
+}
+
+/**
+ * Hook to manage paginated search results from GraphQL query
+ */
+export const usePaginatedResults = ({
+  cql,
+  hasQuery,
+  onShelf,
+  pageSize
+}: UsePaginatedResultsProps): UsePaginatedResultsReturn => {
+  const [resultItems, setResultItems] = useState<Work[]>([]);
+  const [hitcount, setHitCount] = useState(0);
+  const [isRefetching, setIsRefetching] = useState(false);
+  const [lastQueryStr, setLastQueryStr] = useState("");
```

Comment:
Perhaps we could replace this with [`usePrevious()`](https://github.com/streamich/react-use/blob/master/docs/usePrevious.md).